### PR TITLE
use contract versioning

### PR DESCRIPTION
**Pull request by thegostep**

### CODE REVIEW

File: contracts/DaiBackstopSyndicateV1.sol [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+
+    // Combine Dai locked in auctions with the balance on the contract.
+    uint256 combinedDai = daiLockedInAuctions.add(daiBalance);
+
+    // Determine the Maker currently held by the contract.
+    uint256 makerBalance = _MKR.balanceOf(address(this));
+
+    // Determine the amount of Dai and MKR to redeem based on the share.
+    daiRedeemed = combinedDai.mul(shareFloat) / 1e18;
+    mkrRedeemed = makerBalance.mul(shareFloat) / 1e18;
```
Comment (thegostep) [inline contracts/DaiBackstopSyndicateV1.sol]: use .div() ?

File: contracts/DaiBackstopSyndicateV1.sol [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+    // Determine the Dai currently held by the contract.
+    uint256 daiBalance = _DAI.balanceOf(address(this));
+
+    // Jump straight to deactivated if there is not enough dai to make auctions.
+    if (daiBalance < 50000 * 1e18) {
+      _status = Status.DEACTIVATED;
+    } else {
+      // Set the status as active.
+      _status = Status.ACTIVATED;
+    }
```
Comment (thegostep) [inline contracts/DaiBackstopSyndicateV1.sol]: Why do we deactivate here? If we lose an auction, do we get the Dai back? In which case we would need to re-activate

File: contracts/DaiBackstopSyndicateV1.sol (hunk 1 above)
Comment (0age) [reply contracts/DaiBackstopSyndicateV1.sol]: `.div()` just checks that the divisor is not zero - no need with constants

File: contracts/DaiBackstopSyndicateV1.sol (hunk 2 above)
Comment (0age) [reply contracts/DaiBackstopSyndicateV1.sol]: deactivation doesn't prevent withdrawals (you can always withdraw as long as the funds aren't being used in an auction), only entering new auctions

Comment (PhABC) [reply contracts/DaiBackstopSyndicateV1.sol]: If we hit this point, we are basically saying that the backstop failed and everything is aborded?

Comment (PhABC) [reply contracts/DaiBackstopSyndicateV1.sol]: I feel like changing it to ;

```solidity
if (daiBalance < 50000 * 1e18) {
  _status = Status.Status.ACCEPTING_DEPOSITS;
}
```

Would be more powerful and and secure. We could even deploy new instances of this contract while the auction is live and people could try to get to 50k then participate in auction. Doesn't allow for sandwiches either.

Comment (PhABC) [reply contracts/DaiBackstopSyndicateV1.sol]: Which makes me think, what if we have 40k DAI in and Bob deposits 10k DAI, activate it and then withdraw its 10k *after* it was activated. What happens?

**Pull request closed without merging**